plugins/topology/src/utils/resource-utils.ts

```diff
@@ -84,10 +84,8 @@ const getPodTemplate = (
   switch (resource.kind) {
     case 'Pod':
       return resource as V1PodTemplate;
-    case 'Deployment':
-      return (resource as V1Deployment).spec?.template;
     default:
-      return undefined;
+      return resource.spec?.template;
```

Comment:
@debsmita1 shouldn't this be different for CronJobs? Something like `resource.spec?.jobTemplate?.spec?.template`. Can you confirm this and make the changes accordingly as otherwise we wont see the decorator on CJs.

Reply:
Also instead of adding it in default we can be specific and add separate cases for each workload similar to `Deployment`.

Reply:
Updated